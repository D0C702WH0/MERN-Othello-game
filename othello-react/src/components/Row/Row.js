import React from 'react'
import Square from '../Square/Square'
import PropTypes from 'prop-types'
import './Row.css'

export default function Row ({ col, click }) {
  return (
    < div className="row">
      {col.map((square, i) => (
        <Square
          key={i}
          rowI={square._rowIndex}
          colI={square._colIndex}
          value={square._pieceType}
          click={click} />))}
    </div >
  )
}

Row.propTypes = {
  col: PropTypes.array,
  click: PropTypes.func
}
