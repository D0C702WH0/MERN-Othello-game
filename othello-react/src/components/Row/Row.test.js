import React from 'react'
import TestRenderer from 'react-test-renderer'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Row from './Row'
import reversi from 'reversi'

const GameGameInstance = reversi.Game
const game = new GameGameInstance()

configure({ adapter: new Adapter() })

const col = game._board._squares[4]
const colAfterBlackMove = [
  { _pieceType: 'BLANK', _rowIndex: 4, _colIndex: 0 },
  { _pieceType: 'BLANK', _rowIndex: 4, _colIndex: 1 },
  { _pieceType: 'BLANK', _rowIndex: 4, _colIndex: 2 },
  { _pieceType: 'BLACK', _rowIndex: 4, _colIndex: 3 },
  { _pieceType: 'WHITE', _rowIndex: 4, _colIndex: 4 },
  { _pieceType: 'BLACK', _rowIndex: 4, _colIndex: 5 },
  { _pieceType: 'BLANK', _rowIndex: 4, _colIndex: 6 },
  { _pieceType: 'BLANK', _rowIndex: 4, _colIndex: 7 }
]

const click = jest.fn()
let renderer
beforeEach(() => {
  renderer = mount(
    <Row
      col={col}
      click={click} />
  )
})

describe('Row', () => {
  it('Basic rendering', () => {
    const renderer = TestRenderer.create(
      <Row click={click} col={col}/>)
    const result = renderer.toJSON()
    expect(result).toMatchSnapshot()
  })

  it('Rendering after black move', () => {
    const renderer = TestRenderer.create(
      <Row click={click} col={colAfterBlackMove}/>)
    const result = renderer.toJSON()
    expect(result).toMatchSnapshot()
  })

  it('should render two black pawns after move', () => {
    renderer.setProps({ col: colAfterBlackMove })
    expect(renderer.find('.black')).toHaveLength(2)
  })
})
