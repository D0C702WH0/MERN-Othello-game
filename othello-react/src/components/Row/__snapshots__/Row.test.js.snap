// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Basic rendering 1`] = `
<div
  className="row"
>
  <div
    className="square"
    onClick={[Function]}
  >
    <div
      className="pawn undefined"
    />
  </div>
  <div
    className="square"
    onClick={[Function]}
  >
    <div
      className="pawn undefined"
    />
  </div>
  <div
    className="square"
    onClick={[Function]}
  >
    <div
      className="pawn undefined"
    />
  </div>
  <div
    className="square"
    onClick={[Function]}
  >
    <div
      className="pawn black"
    />
  </div>
  <div
    className="square"
    onClick={[Function]}
  >
    <div
      className="pawn white"
    />
  </div>
  <div
    className="square"
    onClick={[Function]}
  >
    <div
      className="pawn undefined"
    />
  </div>
  <div
    className="square"
    onClick={[Function]}
  >
    <div
      className="pawn undefined"
    />
  </div>
  <div
    className="square"
    onClick={[Function]}
  >
    <div
      className="pawn undefined"
    />
  </div>
</div>
`;
